test/replaytests/continue_as_new_wf.go

```diff
@@ -0,0 +1,26 @@
+package replaytests
+
+import (
+	"go.uber.org/cadence/workflow"
+	"go.uber.org/zap"
+)
+
+// ContinueAsNewWorkflow is a sample Cadence workflows that can receive a signal
+func ContinueAsNewWorkflow(ctx workflow.Context) error {
+	selector := workflow.NewSelector(ctx)
+	var signalResult string
+	signalName := "helloWorldSignal"
+	for {
+		signalChan := workflow.GetSignalChannel(ctx, signalName)
+		selector.AddReceive(signalChan, func(c workflow.Channel, more bool) {
+			c.Receive(ctx, &signalResult)
+			workflow.GetLogger(ctx).Info("Received age signalResult from signal!", zap.String("signal", signalName), zap.String("value", signalResult))
+		})
+		workflow.GetLogger(ctx).Info("Waiting for signal on channel.. " + signalName)
+		// Wait for signal
+		selector.Select(ctx)
+		if signalResult == "kill" {
+			return nil
+		}
+	}
+}
```

Comment:
since the only history this is running on is the `continue_as_new.json`, this could be just
```go
func ContinueAsNewWorkflow(ctx workflow.Context) error {
	return workflow.GetSignalChannel(ctx, "unused").Receive(ctx, nil)
}
```
and it would be exactly equivalent.

Reply:
afaik, continue ask new cases fail for only those scenarios where a workflow was marked continue as new and then the history has nothing after continue as new decision. 
This looks like an attempt to cover the edge case where sometimes the workflow continues as new after receiving a signal. the continue as new case has been pretty inconsistent to begin with.

I would say let's keep the original behavior here to preserve the original scenario where we first observed this bug.

Reply:
there's no signal in the history though, so it can't be a "received one, did not receive a second" kind of test.  anything that waits forever without recording an event should be equivalent (signal chan, regular chan, waitgroup, etc all would work)